src/CiaoService.ts

```diff
@@ -882,6 +901,16 @@ export class CiaoService extends EventEmitter {
         aaaaUniqueLocalRecordMap[name] = new AAAARecord(this.hostname, networkInterface.uniqueLocalIpv6, true);
         reverseAddressMap[networkInterface.uniqueLocalIpv6] = new PTRRecord(formatReverseAddressPTRName(networkInterface.uniqueLocalIpv6), this.hostname);
       }
+
+      if (this.overrideAdvertised) {
+        if (net.isIPv4(this.overrideAdvertised)) {
+          aRecordMap[name] = new ARecord(this.hostname, this.overrideAdvertised, true);
+          reverseAddressMap[this.overrideAdvertised] = new PTRRecord(formatReverseAddressPTRName(this.overrideAdvertised), this.hostname);
+        } else if (net.isIPv6(this.overrideAdvertised)){
+          aaaaUniqueLocalRecordMap[name] = new AAAARecord(this.hostname, this.overrideAdvertised, true);
+          reverseAddressMap[this.overrideAdvertised] = new PTRRecord(formatReverseAddressPTRName(this.overrideAdvertised), this.hostname);
+        }
+      }
```

Comment:
You documented the option as "If specified, the mDNS server will advertise this address only.". This is not the case.
It currently just adds the address alongside the other addresses of the interface (or addresses filter via `restrictedAddresses` option). So it doesn't "override" or "replace" anything. Am I missing something here?


Also, are we handling the loopback interface somewhere? E.g., how should a localhost querier see the service? We definitely should keep the localhost addresses (e.g., 127.0.0.1 or ::1). But exposing the "external" docker address would also be unexpected within the container itself?
Would it make sense then to restructure the "overrideAdvertised" address a bit, to explicitly specify the interface for which the address should be used (e.g., `["en0": "192.168.1.27"]`)? This would also allow to specify different override addresses for each an every interface?
What is you opinion here? Also don't want to make it to complex.

Reply:
If you specify an IPv4 it'll set that for all interfaces. If the interface has an IPv6 address that one is left alone.

Equally if you put in an IPv6 address it'll set it but leave the IPv4 address alone. 

And you're right, there is a mismatch here as we don't override all addresses, only the type (IPv4 or IPv6) of the one specified.

After reading your comments my first thought was to update the setting to something like this: `{ [if: string]: IPAddress } | IPAddress`

In both cases we would ignore `lo` as seemingly there is no point to override the advertised address for it. 

Now this only allows you to override either IPv4 or IPv6. We could expand it to something like:

```
type IfIPAddress = IPv4Address | IPv6Address | [IPv4Address, IPv6Address] | [IPv6Address, IPv4Address]


/**
 * Service options supplied when creating a new ciao service.
 */
export interface ServiceOptions {
    // ...
    overrideAdvertised?: { [if: string]: IfIPAddress } | IfIPAddress
```

Thanks!

Reply:
I think the updated type as explained in https://github.com/homebridge/ciao/pull/40#issuecomment-2325097954 make sense.

> If you specify an IPv4 it'll set that for all interfaces. If the interface has an IPv6 address that one is left alone.
> 
> Equally if you put in an IPv6 address it'll set it but leave the IPv4 address alone.
> 
> And you're right, there is a mismatch here as we don't override all addresses, only the type (IPv4 or IPv6) of the one specified.
>
> In both cases we would ignore `lo` as seemingly there is no point to override the advertised address for it.

Sounds good 👍